src/mapdata.h

```diff
@@ -395,26 +401,31 @@ struct submap {
     // Signage is a pretend union between furniture on a square and stored
     // writing on the square. When both are present, we have signage.
     // Its effect is meant to be cosmetic and atmospheric only.
-    inline bool has_signage(int x, int y) {
+    inline bool has_signage(int x, int y) const {
         furn_id f = frn[x][y];
-        if (furnlist[f].id == "f_sign") {
+        if( furnlist[f].id == "f_sign" ) {
             return cosmetics[x][y].find("SIGNAGE") != cosmetics[x][y].end();
         }
+
         return false;
     }
     // Dependent on furniture + cosmetics.
-    inline const std::string get_signage(int x, int y) {
-        if (has_signage(x, y)) {
-            return cosmetics[x][y]["SIGNAGE"];
+    inline const std::string get_signage(int x, int y) const {
+        furn_id f = frn[x][y];
+        if( furnlist[f].id == "f_sign" ) {
+            return cosmetics[x][y].find("SIGNAGE")->second;
```

Comment:
You should check the iterator returned by `find`, it's quite possible to only have a `f_sign` there but no entry in the cosmetics map. Mapgen (especially json based) allows this.